app/javascript/components/project-submissions/components/__tests__/create-form.test.jsx

```diff
@@ -50,6 +50,39 @@ describe('CreateForm', () => {
 
       expect(onClose).toHaveBeenCalledTimes(1);
     });
+
+  });
+
+  describe('When submissions have whitespace.', () => {
+    beforeEach(async() => {
+      render(
+        <ProjectSubmissionContext.Provider value={{ lesson: { has_live_preview: true } }}>
+          <CreateForm
+            onSubmit={onSubmit}
+            userId={userId}
+            onClose={onClose}
+          />
+          ,
+        </ProjectSubmissionContext.Provider>,
+      );
+        
+      await act(async () => {
+        const repoUrlInput = screen.getByTestId('repo-url-field');
+        const livePreviewUrlInput = screen.getByTestId('live-preview-url-field');
+        const isPublicToggle = screen.getByTestId('is-public-toggle-slider');
+
+        fireEvent.change(repoUrlInput, { target: { value: '    https://github.com/repo-url      ' } });
+        fireEvent.change(livePreviewUrlInput, { target: { value: '     https://live-preview.com     ' } });
+        fireEvent.change(isPublicToggle, { target: { checked: false } });
```

Comment:
Nit, sorry @linkonsat, just one last thing. I think we should remove this event for the public toggle since it's not needed. Just having the necessary events in the setup phase makes the test easier to read and will let future readers know what's really important in the test.

Reply:
Yeah only including the bare minimum of what you need is the way to go, it reduces "noise" in the tests by a ton. It also makes things a whole lot more changeable. For example, if we were to rename or remove the public toggle in this component, we'd also have to have to update this unrelated test. On its own, it doesn't seem like a lot, but it compounds with each and every test added to the system.

Reply:
Sweet! I'm definitely going to carry that forward and it's definitely making me rethink my testing approach so far. Since I definitely do see the advantage in getting rid of that compounding redundancy. I'll have to add that to the doc!